### PR TITLE
QOL-7939 prepare for CKAN 2.9

### PR DESCRIPTION
**Pull request by ThrawnCA**
- update Continuous Integration test config to handle Click CLI
- add Flask routing to match Pylons

### CODE REVIEW

**Comment by ThrawnCA** [comment]
Needs other extensions like ckanext-validation to be updated before CKAN 2.9 can pass

**Review by duttonw — APPROVED** [review]

**Review by duttonw — APPROVED** [review]

**Review by duttonw — APPROVED** [review]

**Review by duttonw — APPROVED** [review]

File: ckanext/data_qld/controller_functions.py [hunk 1]
```diff
@@ -75,3 +81,368 @@ def show_schema(dataset_id, resource_id):
     except tk.NotAuthorized as e:
         log.warn(e)
         return tk.abort(403, _('You are not authorized to view the Data Scheme for the resource %s' % resource_id))
+
+
+def _valid_report_type(report_type):
```
Comment (duttonw) [inline ckanext/data_qld/controller_functions.py:86]: can you put the 'report' controller functions into a new class to seperate out and give context that this is 'extensions' to the report plugin etc.

Comment (duttonw) [reply ckanext/data_qld/controller_functions.py:86]: .. seems your merged them, why is that?

File: test/features/resource_availability.feature [hunk 2]
```diff
@@ -165,44 +167,15 @@ Feature: Re-identification risk governance acknowledgement or Resource visibilit
...
+        And I should not see "Re-identification risk governance acknowledgement/Resource visibility"
 
         Examples: Users
             | User          |
             | SysAdmin      |
             | TestOrgAdmin  |
             | TestOrgEditor |
-
-
-    Scenario Outline: An editor, admin or sysadmin user views a dataset resource additional info
```
Comment (duttonw) [inline test/features/resource_availability.feature:171]: why delete dataset from here?

Comment (duttonw) [reply test/features/resource_availability.feature:171]: nm, the steps are in tests above.

Comment (duttonw) [reply test/features/resource_availability.feature:171]: please updated tests which were extended 
Scenario Outline: User creates resource for a dataset with Contains de-identified data is NO
to better match what it now does

like

Scenario Outline: User creates resource for a dataset with Contains de-identified data is NO and verifies it being swapped and not seeing settings etc etc

File: ckanext/data_qld/fanstatic/flag-comment.js [hunk 3]
```diff
@@ -46,6 +46,7 @@ this.ckan.module('flag-comment', function (jQuery) {
      * Returns nothing.
      */
     initialize: function () {
+      jQuery('#comment_form').append("DEBUG: flag-comment.js:initialize");
```
Comment (duttonw) [inline ckanext/data_qld/fanstatic/flag-comment.js:49]: do you still want debug in this repo?

**Review by duttonw — APPROVED** [review]

File: ckanext/data_qld/controller_functions.py (hunk 1 above)
Comment (ThrawnCA) [reply ckanext/data_qld/controller_functions.py:84]: Hadn't seen this comment, but yes, I can split them.

File: ckanext/data_qld/fanstatic/flag-comment.js (hunk 3 above)
Comment (ThrawnCA) [reply ckanext/data_qld/fanstatic/flag-comment.js:49]: I'm still encountering some issues with the JavaScript on the flagging test, just for CKAN 2.9, so I'll keep it for now, but I should delete this before merging.

**Pull request merged**